includes/class-sensei-lesson.php

```diff
@@ -1116,8 +1115,13 @@ private function save_post_meta( $post_key = '', $post_id = 0 ) {
 			$new_meta_value = '-1';
 		}
 
+		// If course is empty, unassign from course.
+		if ( 'lesson_course' === $post_key && empty( $new_meta_value ) ) {
+			$new_meta_value = '';
```

Comment:
This assignment is not needed. If the user selects 'None' it is going to be an empty string anyway.

Reply:
Yeah you are right!  The check added below is enough to fix this.